src/Network/Mattermost/Types.hs

```diff
@@ -89,11 +92,23 @@ createPool host port ctx cpc connTy =
   Pool.createPool (mkConnection ctx host port connTy >>= newMMConn) closeMMConn
                   (cpStripesCount cpc) (cpIdleConnTimeout cpc) (cpMaxConnCount cpc)
 
-initConnectionData :: Hostname -> Port -> T.Text -> Maybe BasicAuth -> ConnectionType -> ConnectionPoolConfig -> IO ConnectionData
-initConnectionData host port path ba connTy cpc = do
+initConnectionData :: Hostname -> Port -> T.Text -> ConnectionType -> ConnectionPoolConfig -> IO ConnectionData
+initConnectionData host port path connTy cpc = do
   ctx  <- initConnectionContext
   pool <- createPool host port ctx cpc connTy
-  return (mkConnectionData host port path pool connTy ctx ba)
+  return (mkConnectionData host port path pool connTy ctx)
+
+setConnectionRequestTransformation :: (HTTP.Request_String -> HTTP.Request_String) -> ConnectionData -> ConnectionData
+setConnectionRequestTransformation trans cd =
+  cd
+    { cdConnReqTrans = trans
+    }
+
+setWsRequestTransformation :: (WS.Headers -> WS.Headers) -> ConnectionData -> ConnectionData
+setWsRequestTransformation trans cd =
+  cd
+    { cdWsReqTrans = trans
```

Comment:
Just a style note: please format this with the opening brace on the same line as the record value, i.e.,
```
cd { cdWsReqTrans ...
   }
```

Reply:
Ok, changed it.

Reply:
Thanks!